### ISSUE TITLE
Can't replace friends/chat UI

### ISSUE THREAD

**Issue by Evity** [PhantomGamers/SteamFriendsPatcher#10]
library UI replaced successfully but friends/chat UI fails to replace.

**Comment by PhantomGamers** [comment]
You have to keep pressing force check until it works, it likely means your Steam client is using a different version of the css than the one the patcher is looking for

**Comment by PhantomGamers** [comment]
Also Steam just pushed a new version of the friends css a few hours ago so restart Steam before trying to patch if you haven't

**Comment by lucaboox** [comment]
I've restarted and pressed it about like 50 times and it still won't work sometimes it's so random

**Comment by PhantomGamers** [comment]
> I've restarted and pressed it about like 50 times and it still won't work sometimes it's so random

Yep, nothing I can really do about it for now, it's completely rng :/

For me it still works 99% of the time, and then when it doesn't just one or two force checks get it to work.

**Comment by lucaboox** [comment]
> > I've restarted and pressed it about like 50 times and it still won't work sometimes it's so random
> 
> Yep, nothing I can really do about it for now, it's completely rng :/
> 
> For me it still works 99% of the time, and then when it doesn't just one or two force checks get it to work.

Ya I've got it to work actually I think I was just doing it weird, just glad I can get a skin I like. Thanks for making this 100% it's so great lol.

**Comment by ghost** [comment]
> Ya I've got it to work actually I think I was just doing it weird, just glad I can get a skin I like. Thanks for making this 100% it's so great lol.

What'd you do to get it to work? Mine is still pretty inconsistent and rather irksome.

**Comment by PhantomGamers** [comment]
Yeah I'd love to know if you did something differently when you got it to work

**Comment by yurisuika** [comment]
Recently this has been nearly impossible to get to patch... I don't know if I am really seeing a pattern, but here is what I've last done and had it successfully patch the chat. Patcher is set to restart Steam after patch. Often I will see the friends.css showing it is patched and now imports ofriends.custom.css and ofriends.original.css, but nothing actually has changed in the windows... Not sure if that is the old patcher or this one that is the culprit there. Anyways...

- ...after lots of failed attempts with both patchers, I get to a point where I close all related processes.
- Run Steam.
- Patcher running in background from startup will fix up the library, but not chat.
- Potentially clear cache, restarting Steam. See chat is still not patched, library css folder is wiped as of 0.1.36, so copy that backup over.
- Exit Steam.
- End patcher process.
- Run Steam.
- Run old chat patcher. Shows green success.
- Exit Steam.
- Run Steam. Chat is themed.

For example after this success I then exited Steam and ran it again and chat was back to default. I run this patcher again (which the process was no longer running), and then it fixes up the chat. I then exit Steam again and run again, but this time chat is fixed up right from the get-go. I exit Steam again and run again and it is no longer themed. I run the patcher again and then it goes into a restart loop starting Steam up 10-20 times, chat is not themed after it settles. I run the patcher again (still not having killed the process like before in task manager), and it restarts Steam once and chat is themed.

At least recently, it seems that I can't get the chat to theme ever until I get one successful patch with the old chat patcher. I know this was brought up somewhere else in an issue here, I guess they work differently.

**Comment by PhantomGamers** [comment]
Sadly the issue seems to be on Steam's end, it seems either the cloudflare cdn or the akamaihd variant of the friends.css just refuse to load from cache, so steam grabs a new one whenever the window is reloaded. I'm unsure which one. When it works, it's because Steam loaded the variant that it *can* read from cache.

Instead of restarting Steam a bunch, you can also just sign out and then back into the friends list as this also triggers the friends window to recheck for a new css file (or, if it *can* read from the cache, it'll load the patched version from cache). Note that depending on your luck, it may just keep grabbing the broken version... it's less than ideal. I don't really know how to fix this though.

(This is all mostly theoretical as I have no real familiarity with the workings of CEF, which is what Valve is using for the new UIs)

Also as for this
> Potentially clear cache, restarting Steam. See chat is still not patched, library css folder is wiped as of 0.1.36, so copy that backup over.  

I've done this because of Valve deciding to randomly name the libraryroot.css it uses in the beta version of Steam to "5.css" which makes it harder to track which files need to be restored. I know how to fix this but I'm busy with finals and don't have time to do it right now, so just deleting the folder and letting steam reextract everything seemed like the simpler approach.

**Comment by yurisuika** [comment]
It has actually been incredibly consistent in the past several days for me, surprisingly just after I commented.

I have it set to startup. I load Steam and it is not patched, so I open the patcher and clear cache. Replace the cleaned library files, and it is all set.

I've been dealing with a program locking up my PC often over the past week, so I've been having to patch Steam many, many times a day, and since maybe the 13th it has only once not easily gone through the aforementioned process.

Thanks for your efforts in combating Steam's killing of skinning! PETA must have gotten confused and gone after Valve's arse.

**Comment by jakiisheep** [comment]
I have this problem too. Sometimes it just patches and the next day when I wake up and turn on my pc chat is not themed. Really weird. The library works every time but friends not. I guess I won't theme friends anymore until I find a solution.

**Comment by mutageneral** [comment]
I was able to get the friends working, but then I changed the CSS again, and I am not able to get it working again :(

Also is it normal that sometimes the scanning function for the friend UI runs in a loop? If I let it run for a long time will it end up being able to patch the friend UI? I did let it run for multiple minutes, but it did not patch the friend UI.

Edit: I just find out the patching loop thing was an unrelated bug #14 and that if I spammed the force check button for a while it was able to patch the friend UI and that's probably what I did end up doing the first time. Weird that the other loop bug didn't end up being able to patch it as it was probably also patching over and over again.
